File: app/tests/hand.py
class Hand:
    MAX_HAND_SIZE = 10

    def __init__(self):
        self.cards = []

    def add_card(self, card):
        if len(self.cards) >= self.MAX_HAND_SIZE:
            raise ValueError("Cannot add more cards to hand. Maximum size reached.")
        self.cards.append(card)

    def play_card(self, index):
        card_to_play = self.remove_card(index)
        return card_to_play
    
    def remove_card(self, index):
        if index < 0 or index >= len(self.cards):
            raise IndexError("Invalid card index.")
        return self.cards.pop(index)
    
    def clear_hand(self):
        self.cards.clear()

    @property
    def count(self):
        return len(self.cards)
    
    def __str__(self):
        return "\n".join(str(card) for card in self.cards)

File: app/classes/Hand.cs
public class Hand {
    private List<Card> cards;
    private const int MaxHandSize = 10;

    public Hand() {
        cards = new List<Card>();
    }

    public void AddCard(Card card) {
        if (cards.Count >= MaxHandSize) {
            throw new InvalidOperationException("Cannot add more cards to hand. Maximum size reached.");
        }
        cards.Add(card);
    }

    public Card PlayCard(int index) {
        if (index < 0 || index >= cards.Count) {
            throw new ArgumentOutOfRangeException("Invalid card index.");
        }
        Card cardToPlay = cards[index];
        cards.RemoveAt(index);
        return cardToPlay;
    }

    public void ClearHand() {
        cards.Clear();
    }

    public int Count => cards.Count;

    public override string ToString() {
        return string.Join("\n", cards);
    }
}